development/kops/delete_store.sh

```diff
@@ -0,0 +1,37 @@
+#!/usr/bin/env bash
+# Copyright 2020 Amazon.com Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+set -eo pipefail
+export KOPS_STATE_STORE=${1:-${KOPS_STATE_STORE}}
+if [ -z "${KOPS_STATE_STORE}" ]
+then
+  echo "Usage: ${0} s3://bucketname"
+  echo "  or set and export KOPS_STATE_STORE"
+  exit 1
+fi
+if [[ "${KOPS_STATE_STORE}" != s3://* ]]
+then
+  export KOPS_STATE_STORE="s3://${KOPS_STATE_STORE}"
+fi
+unset KOPS_CLUSTER_NAME
+echo "Deleting kops store $KOPS_STATE_STORE"
+if kops get cluster --state "${KOPS_STATE_STORE}" 2>/dev/null
+then
+  echo "Delete the kops cluster before deleting the store"
+  exit 1
+fi
+BUCKET_NAME=$(echo "${KOPS_STATE_STORE}" | sed -e 's,s3://,,g')
+set -x
+aws s3 rm --recursive "${KOPS_STATE_STORE}" || true
```

Comment:
What happens when this call fails? Will we be able to delete the underlying bucket?

Reply:
This is script is supposed to be like `rm -rf bucket` and so I am ignoring the return on this so that we always try to remove the bucket.

We can only remove an empty bucket. If the content removal fails, it is likely the bucket removal will fail.